packages/sdk-bridge/src/backend.ts

```diff
@@ -0,0 +1,242 @@
+import {  GoldSkyBackend, GoldSkyMessage, MessageFilter } from "@nomad-xyz/sdk";
+import {MessageBackend} from "@nomad-xyz/sdk";
+import { request, gql } from 'graphql-request';
+import * as config from '@nomad-xyz/configuration';
+
+const defaultGoldSkySecret = "";
+
+
+/**
+ * Abstract class required for operation of NomadMessage
+ */
+ export default abstract class BridgeMessageBackend extends MessageBackend {
+    abstract sender(messageHash: string): Promise<string | undefined>;
+    abstract receivedTx(messageHash: string): Promise<string | undefined>;
+}
+
+
+export type GoldSkyBridgeMessage =  GoldSkyMessage & {
+    send_tx: string,
+    sent_at: string,
+    send_block: string,
+    original_sender: string,
+    receive_tx: string,
+    origin_and_nonce: string,
+    receive_block: string,
+    received_at: string,
+};
+
+/**
+ * GoldSky bridge backend for NomadMessage 
+ */
+export class GoldSkyBridgeBackend extends GoldSkyBackend implements BridgeMessageBackend {
+
+    messageCache: Map<string, GoldSkyBridgeMessage>;
+
+    constructor(env: string, secret: string) {
+        super(env, secret);
+        this.messageCache = new Map();
+    }
+
+    /**
+     * Checks whether an environment is supported by the backend. Throws on unsupported
+     * @param environment environment to check
+     */
+    static checkEnvironment(environment: string): void {
+        if (environment != 'production') {
+            throw new Error(`Only production environment is supported`);
+        }
+    }
+
+    /**
+     * Creates a default GoldSky backend for an environment
+     * @param environment environment to create the backend for
+     * @returns backend
+     */
+    static default(environment: string | config.NomadConfig = 'development'): GoldSkyBridgeBackend {
+        const environmentString = typeof environment === 'string' ? environment : environment.environment;
+
+        GoldSkyBridgeBackend.checkEnvironment(environmentString);
+        
+        const secret = process.env.GOLDSKY_SECRET || defaultGoldSkySecret;
+        if (!secret) throw new Error(`GOLDSKY_SECRET not found in env`);
+        return new GoldSkyBridgeBackend(environmentString, secret);
+    }
+
+    /**
+     * Stores message into internal cache
+     */
+    storeMessage(m: GoldSkyBridgeMessage): void {
+        this.messageCache.set(m.message_hash, m);
+        this.dispatchTxToMessageHash.set(m.dispatch_tx, m.message_hash);
+    }
+
+    /**
+     * Get the message representation associated with this message (if any)
+     * by message hash
+     *
+     * @returns A message representation (if any)
+     */
+    async getMessage(messageHash: string): Promise<GoldSkyBridgeMessage | undefined> {
+        let m = this.messageCache.get(messageHash);
```

Comment:
This will not allow for updates in message state.  For example, it sets `messageCache` initially, and then it is relayed. We won't get the relay information

Reply:
I guess we could use `forceFetch` with `true` for this. But yeah, I wonder if having the cache might cause more confusion than it's worth. What do you think @kekonen / @ErinHales ?